src/scaffold/app/nix.rs

```diff
@@ -9,47 +9,76 @@ use crate::error::{ScaffoldError, ScaffoldResult};
 use crate::file_tree::*;
 use crate::versions::holochain_nix_version;
 
-pub fn flake_nix() -> FileTree {
+pub fn flake_nix(holo_enabled: bool) -> FileTree {
     let holochain_nix_version = holochain_nix_version();
-    file!(format!(
+
+    // Define the holo-specific parts
+    let additional_packages = if holo_enabled {
+        "pkgs.curl\n                    # more packages go here"
+    } else {
+        "# more packages go here"
+    };
+
+    let extra_content = if holo_enabled {
+        r#"extraSubstitutors = [ "https://cache.holo.host" ];
+                trustedPublicKeys = [
+                    "cache.holo.host-2:ZJCkX3AUYZ8soxTLfTb60g+F3MkWD7hkH9y8CgqwhDQ="
+                ];
+
+                shellHook = ''
+                    nix-env -f "https://hydra.holo.host/channel/custom/holo-nixpkgs/alpha/holo-nixpkgs/nixexprs.tar.xz" -iA holo-dev-server
```

Comment:
I think this is potentially surprising for updated versions of a binary to be installed on starting a shell. Is there another way of doing this so it's included as a package and locked with the flake?

Reply:
Should it be locked with the flake? This binary should represent production Holo Network, and that can change.

Reply:
That said, adding it in this way seems ugly to me. I just don't know a better way

Reply:
After discussing I understand why this is hard to change. With the source code being closed source it's not simple to build from source and hydra is giving us only the latest binary per channel so we don't have an obvious way to lock to a specific version. It would be good to solve this at some point so that people can look at the flake.lock and see what version they have, and when it gets updated. This also makes offline development difficult but for now the install line could be commented out temporarily and the binary manually installed before going offline or some other workaround.

Reply:
this alters the user's profile permanently and not just for the duration of the nix shell. what is the chance to open-source the holo-dev-server?

Reply:
eek

Reply:
packaging closed-source binaries isn't great but it's feasible. it does require some form locking, which can be automatically maintained on a separate public repo and consumed here. given that the scaffolding performs a `nix flake update` makes the consumption of it a non-issue.